slack/rtm/client.py

```diff
@@ -433,37 +447,27 @@ async def _dispatch_event(self, event, data=None):
                     # close/error callbacks.
                     break
 
-                if inspect.iscoroutinefunction(callback):
+                if self.run_async or inspect.iscoroutinefunction(callback):
                     await callback(
                         rtm_client=self, web_client=self._web_client, data=data
                     )
                 else:
-                    self._execute_in_thread(callback, data)
+                    await self._execute_in_thread(
```

Comment:
`_execute_in_thread` is also a coroutine so that the behavior in both async and sync is consistent here.

Reply:
this change makes handling errors consistent for both `run_async=True` and `False`. This addresses both #530 and #613

Reply:
>The same behavior could be accomplished by simply running the callback on the current thread, right?

This is true here. I'll check if we can get rid of the thread pool (I think it's feasible).